src/services/formatting/smartIndenter.ts

```diff
@@ -268,6 +275,16 @@ namespace ts.formatting {
             return sourceFile.getLineAndCharacterOfPosition(n.getStart(sourceFile));
         }
 
+        export function isArgumentAndStartLineOverlapsExpressionBeingCalled(parent: Node, child: Node, childStartLine: number, sourceFile: SourceFileLike): boolean {
+            if (!(isCallExpression(parent) && contains(parent.arguments, child))) {

```

Comment:
Would a similar issue arise for an index expression in an array access?

Reply:
We could keep looking into awkward situations ad infinitum, but we would increasingly approach patterns that don't resemble code user write. To avoid a rabbit hole, I'd prefer to keep this change conservative and get it in. If we run into issues in the course of use or get customer feedback, we can adjust our indentation then.

This change was designed to minimally effect indentation except for the args-in-iife case described in the issue. The supplementary tests add (which pass unchanged from master) provide some confirmation this actually holds.